homeassistant/components/sensor/dte_energy_bridge.py

```diff
@@ -0,0 +1,82 @@
+"""Support for monitoring energy usage using the DTE energy bridge."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:flash'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the DTE energy bridge sensor."""
+    ip_address = config.get('ip')
+    if not ip_address:
+        _LOGGER.error(
+            "Configuration Error"
+            "'ip' of the DTE energy bridge is required")
+        return None
+    dev = []
+    dev.append(
+        DteEnergyBridgeSensor(ip_address,
+                              "Current Energy Usage", "instantaneousdemand",
+                              "kW"))
+    add_devices(dev)
+
+
+# pylint: disable=too-many-instance-attributes
+class DteEnergyBridgeSensor(Entity):
+    """Implementation of an DTE Energy Bridge sensor."""
+
+    def __init__(self, ip_address, name, endpoint, unit_of_measurement):
+        """Initialize the sensor."""
+        self._url = "http://{}/".format(ip_address)
+        self._name = name
+        self._endpoint = endpoint
+        self._unit_of_measurement = unit_of_measurement
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of th sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the energy usage data from the DTE energy bridge."""
+        import requests
+        try:
```

Comment:
There is a whole lot inside of this try block. Could some of this code be moved out of the try, except? I know the requests.get should definitely be inside. However, the parsing of the response should probably be outside of the try block.